metadata.yaml

```diff
@@ -0,0 +1,19 @@
+name: training-operator
+display-name: Training Operator
+summary: Tools for distributed or non-distributed training on Kubernetes
+description: |
+  Training Operator provides Kubernetes custom resources that make it
+  easy to run distributed or non-distributed training jobs on Kubernetes.
+  This charm deploys the Operator configured for use with Kubeflow to
+  Kubernetes models in Juju.
+maintainers: [Juju Developers <juju@lists.ubuntu.com>]
+tags: [ai, bigdata, kubeflow, machine-learning, tensorflow, pytorch]
+containers:
+  training-operator:
+    resource: training-operator-image
+resources:
+  training-operator-image:
+    type: oci-image
+    description: OCI image for training-operator
+    auto-fetch: true
```

Comment:
This field is deprecated, technically so is `upstream-source`, though there are benefits to including that for testing purposes.

```suggestion
```

Reply:
I'll remove both keys from now and add `upstream-source` later if I need it for testing.

Reply:
`upstream-source` is in a weird state that I wouldn't call deprecated exactly. I opened this bug to talk about adding official Juju support for this or a similar field:

https://bugs.launchpad.net/juju/+bug/1946121

At any rate, for Kubeflow charms, a lot of our tooling uses `upstream-source`, so we should probably keep it in for now.

Reply:
Alright, keeping it then.